### PR TITLE
Love: Card to show entites on picture

### PR DESCRIPTION
**Pull request by c727**
fix: https://github.com/home-assistant/ui-schema/issues/52

unbenannt

```yaml
  - name: new cards
    cards:
      - type: picture-elements
        title: picture elements
        image: /local/floorplan.png
        elements:
          - type: state-badge
            entity: light.ceiling_lights
            style:
              top: 30%
              left: 55%
              '--paper-item-icon-color': green
          - type: state-badge
            action: toggle
            entity: light.ceiling_lights
            style:
              top: 30%
              left: 20%
```

### CODE REVIEW

**Comment by balloob** [comment]
I think that we can remove top and left and just have style be an object that we assign.

```js
function dashToCamelCase(value) { 
  const parts = value.split('-');
  return parts.shift() + parts.map(s => s.substr(0, 1).toUpperCase() + s.substr(1));
}

const el = document.createElement('bla')
Object.keys(elementConfig.style).forEach((key) => {
  el.style[dashToCamelCase(key)] = elementConfig.style[key];
});
```

**Comment by balloob** [comment]
This will translate calls like:

```
style:
  padding-top: 10px;
```

to

```
el.style.paddingTop = '10px'
```

File: src/panels/lovelace/cards/hui-entities-gone-wild.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+  getCardSize() {
+    return 5;
+  }
+
+  _equals(a, b) {
+    return a === b;
+  }
+
+  _computeStateObj(item, states) {
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-entities-gone-wild.js]: `item` -> `entity_id`

Comment (balloob) [reply src/panels/lovelace/cards/hui-entities-gone-wild.js]: Not being used.

**Comment by arsaboo** [comment]
Great substitute for Floorplan :+1: 

Will the bulb itself be clickable (preferred) or do we need the `Light Off` button?

**Comment by balloob** [comment]
So, let's limit the initial types of things that we want to support.

What about we just start with type `state-badge` and it renders the `<state-badge>` web component. Tapping it will toggle the state.

**Comment by balloob** [comment]
Also, let's create the elements by hand, not using the Polymer template engine.

**Comment by c727** [comment]
```yaml
    cards:
      - type: entities-gone-wild
        title: entities-gone-wild
        image: /local/floorplan.png
        entities:
          - type: state-badge
            entity_id: light.ceiling_lights
            style:
              top: 30%
              left: 55%  
```

**Comment by balloob** [comment]
Should it be `entity` to stay consistent?

**Comment by balloob** [comment]
Also should the key be entities? As we might add a service button or something to it? Maybe 'elements'

**Comment by balloob** [comment]
What about name, I like the idea of entity-map but that's confusing with a real map.

**Comment by c727** [comment]
yep, it's hard to find a good name. maybe `picture-elements`?

**Comment by balloob** [comment]
Like that!

File: src/panels/lovelace/cards/hui-entities-gone-wild.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+/*
+ * @appliesMixin EventsMixin
+ * @appliesMixin LocalizeMixin
+ */
+class HuiEntitiesGoneWildCard extends LocalizeMixin(EventsMixin(PolymerElement)) {
+  static get template() {
+    return html`
+    <style>
+      ha-card {
+        padding: 16px;
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-entities-gone-wild.js]: We should drop the padding. Let it be the choice of user if they want to add white space to image?

File: src/panels/lovelace/cards/hui-entities-gone-wild.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        transform: translate(-50%, -50%);
+      }
+      #root .clickable {
+        cursor: pointer;
+        padding: 4px;
+      }
+    </style>
+
+    <ha-card>
+      <div class='header'>
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-entities-gone-wild.js]: Why not use heading prop for ha card

File: src/panels/lovelace/cards/hui-entities-gone-wild.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+  _hassChanged(hass) {
+    this._requiresStateObj.forEach((element) => {
+      const { el, entityId } = element;
+      el.stateObj = hass.states[entityId];
+      el.title = this._computeTooltip(entityId, hass);
+    });
+  }
+
+  _computeTooltip(entityId, hass) {
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-entities-gone-wild.js]: Why not just take in a state obj?

**Review by balloob — APPROVED** [review]
Ok once name updated and comments addressed

File: src/panels/lovelace/cards/hui-entities-gone-wild.js (hunk 2 above)
Comment (balloob) [reply src/panels/lovelace/cards/hui-entities-gone-wild.js]: Hmm now I'm not sure about this anymore..

File: src/panels/lovelace/cards/hui-entities-gone-wild.js (hunk 3 above)
Comment (c727) [reply src/panels/lovelace/cards/hui-entities-gone-wild.js]: copy-pasted from "your" cards :P

**Comment by c727** [comment]
Should be OK now.

I like to add the service function again, maybe like this:
```yaml
type: service
text: turn light off
service:
  domain: light
  service: turn_off
  data:
    entity_id: light.kitchen
```

and the "show state + uom" as text:
```yaml
type: state-text
entity: sensor.temp_livingroom
```

**Pull request merged**